agentcfg/fetch_test.go

```diff
@@ -151,3 +155,175 @@ func mockDoc(sampleRate float64) m {
 		},
 	}
 }
+
+func TestDirectConfigurationPrecedence(t *testing.T) {
+	for _, tc := range []struct {
+		query            Query
+		agentConfigs     []config.AgentConfig
+		expectedSettings map[string]string
+	}{
+		{
+			query: Query{
+				Service: Service{
+					Name:        "service1",
+					Environment: "production",
+				},
+			},
+			agentConfigs: []config.AgentConfig{
+				{
+					Service: config.Service{Name: "", Environment: "production"},
+					Config:  map[string]string{"key1": "val2", "key2": "val2"},
+					Etag:    "def456",
+				},
```

Comment:
maybe also add a block above with only Name set (and matching), showing that both Name & Environment always wins?